### DISCUSSION TITLE
Cannot get Streams-i2s-serial to work on ESP32 with INMP441

### DISCUSSION THREAD

**Discussion by 2e0wkw** [pschatzmann/arduino-audio-tools#1311] [Q&A] [answered]
Hi, I am using an ESP32Dev (ESP32-S) board with an INMP441 mic on the Arduino IDE.
I am trying to use the i2s to Serial plotter example but it produces nothing but the following text:
[I] AudioTypes.h : 104 - sample_rate: 44100
[I] AudioTypes.h : 105 - channels: 2
[I] AudioTypes.h : 106 - bits_per_sample: 16
[I] I2SConfigESP32V1.h : 79 - rx/tx mode: RX_MODE
[I] I2SConfigESP32V1.h : 80 - port_no: 0
[I] I2SConfigESP32V1.h : 81 - is_master: Master
[I] I2SConfigESP32V1.h : 82 - sample rate: 44100
[I] I2SConfigESP32V1.h : 83 - bits per sample: 16
[I] I2SConfigESP32V1.h : 84 - number of channels: 2
[I] I2SConfigESP32V1.h : 85 - signal_type: Digital
[I] I2SConfigESP32V1.h : 87 - i2s_format: I2S_STD_FORMAT
[I] I2SConfigESP32V1.h : 92 - buffer_count:6
[I] I2SConfigESP32V1.h : 93 - buffer_size:512
[I] I2SConfigESP32V1.h : 98 - pin_bck: 14
[I] I2SConfigESP32V1.h : 100 - pin_ws: 15
[I] I2SConfigESP32V1.h : 102 - pin_data: 32
[E] I2SESP32V1.h : 95 - size_t audio_tools::I2SDriverESP32V1::readBytes(void*, size_t)
[I] StreamCopy.h : 143 - StreamCopy::copy  1024 -> 0 -> 0 bytes - in 0 hops

I  know the hardware works as i have other code that can receive from the INMP441 and plot the results.

Your help would be appreciated

**Comment by pschatzmann** [comment]
I did not test the new i2s functionality from the esp32 pre-release extensively since I mainly concentrated on the output side.

If you want to have a stable functionality, I recommend to roll back to the latest official release.

I noticed however, that with the RXTX_MODE, I am getting some data. 
But I am not sure if the received data makes any sense...

**Reply by pschatzmann** [reply, marked as answer]
I committed a correction for 
- RX_MODE: channel creation needs to be mode dependent 
- 24 bit: set mclk_multiple

This should resolve your issue.
